sw/device/silicon_creator/rom/doc/rom_overview_specification.md

```diff
@@ -0,0 +1,272 @@
+# Mask ROM Specification
+- Authors: cfrantz
+- Reviewers: alphan, moidx
+- Date: 2021-06-28
+- Status: Implmented
+
+
+## Overview
+
+This document is a high-level overview of the features and responsibilities of the Mask ROM.
+
+This document is not a comprehensive specification.  When appropriate, it refers to more complete specifications for various mask ROM responsibilities.  These specifications include:
+
+
+
+*   [Signature Verification Module [sigverify]](#sigverify)
+*   [Boot Policy Module [bootpolicy]](#bootpolicy)
+*   [Address space unlock module [unlock]](#unlock)
+*   [Bootstrap module [bootstrap]](#bootstrap)
+*   [Bootstrap decoding](#strapping)
+*   [Shutdown Module [shutdown]](#shutdown)
+*   [Chip-Specific Startup [startup]](#startup)
+*   [Mask ROM Pin Configuration [pinconfig]](#pinconfig)
+*   [Memory Protection [memprotect]](memory_protection.md)
+
+
+## Threat Model
+
+
+## Dependencies
+
+The mask ROM is almost fully self-contained. All of the code in the mask ROM is located in //sw/device/silicon\_creator with a few exceptions.
+
+We permit the following exceptions:
+
+
+
+*   Base utility functions from `//sw/device/lib/base`.
+*   The top\_level memory layout from `//hw/top_earlgrey/sw/autogen`.
+*   Autogenerated register definitions (ie: `alert_handler_regs.h`, `flash_ctrl_regs.h`, etc).
+
+TODO: Once we convert the build system to bazel, we can use bazel’s visibility rules to enforce the dependency restrictions in the build.
+
+
+## Manufacturing Bring-Up
+
+During bring-up while the device is in one of the TEST\_UNLOCKED&lt;n> states, certain countermeasures will not be enabled to facilitate testing and bring-up of the chip.
+
+Notably:
+
+
+
+*   The alert handler will not be configured or enabled in TEST\_UNLOCKED&lt;n>.
+*   TODO: [flash scrambling configuration]
+
+
+## Keys
+
+
+### Silicon Creator Keys
+
+The Silicon Creator keys are provisioned when the silicon creator configures and initializes the signing environment.  This process is currently being designed and the details are beyond the scope of this document.  For now, the publicly available development FPGA keys are used as stand-ins for the silicon creator keys.
+
+
+#### ROM.KEYS-SILICON-CREATOR-PROD
+
+These keys are RSA-3072 public keys with exponent 3 or F4.  The PROD keys are valid in all lifecycle states.
+
+
+#### ROM.KEYS-SILICON-CREATOR-DEV
+
+These keys are RSA-3072 public keys with exponent 3 or F4.  The DEV keys are valid in lifecycle states DEV and RMA.
+
+
+#### ROM.KEYS-SILICON-CREATOR-BRINGUP
+
+These keys are RSA-3072 public keys with exponent 3 or F4.  The BRINGUP keys are valid in lifecycle states TEST\_UNLOCKED&lt;n> and RMA.
```

Comment:
Do you want to update the code to use the name "bringup" or update this doc to use "test" for the sake of consistency?

Reply:
Changed to `TEST`.